### PR TITLE
make `ProductBranding` component clickable

### PR DESCRIPTION
**Pull request by michelengelen**
#### Summary
re-arranged the left portion of global header to allow for the `ProductBranding` to be clickable and open the `ProductSwitcherMenu` as well (alongside the `ProductSwitcher` 'IconButton`)

The `ProductBranding` currently has no visual guidance for it to being a clickable element. It would probably be a good idea to at least give it some kind of `hover` state to make it more clear to the user.

Thoughts @andrewbrown00  ?

#### Ticket Link
[MM-37986](https://mattermost.atlassian.net/browse/MM-37986)

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by andrewbrown00** [comment]
Hey, @michelengelen, I don't see the option to enable the **Global Header** in the account settings or in the system console.

image

**Comment by michelengelen** [comment]
it should be enabled by default through a feature flag, or am I missing something @spirosoik ?

**Comment by andrewbrown00** [comment]
> @andrewbrown00 it should be enabled by default through a feature flag, or am I missing something @spirosoik ?

I thought that was the case too, but I don't see it enabled on this test server 🤔

image

**Comment by spirosoik** [comment]
@michelengelen @andrewbrown00 in spinwicks we are not setting anything we just actually call the provisioner or CWS to create the envs. Let me see what are the default values for provisioner in test

**Comment by spirosoik** [comment]
@michelengelen @andrewbrown00 doesn't seem to have this as default value

**Comment by andrewbrown00** [comment]
> @michelengelen @andrewbrown00 doesn't seem to have this as default value

Hey, @spirosoik, who has to set the `default value`? It's blocking me from reviewing a handful of PR's.

cc @michelengelen

**Comment by spirosoik** [comment]
@andrewbrown00 we are syncing the changes in provisioner to have this, I will ping you when it's done

**Comment by spirosoik** [comment]
@michelengelen @andrewbrown00 is this the env variable? `MM_GLOBALHEADER`

**Comment by spirosoik** [comment]
@michelengelen @andrewbrown00 what exactly do we want to see? The global header is forced by Split as enabled

**Comment by michelengelen** [comment]
ok @spirosoik i just tested CWS and the feature flag is not set there as well ... this is the feature flag we set during local server init: `MM_FEATUREFLAGS_GLOBALHEADER=true`

**Comment by spirosoik** [comment]
@michelengelen Split should force it to enable as I see in the test servers. Are we sure that split and global header work as expected?

**Comment by michelengelen** [comment]
@spirosoik I will check that

**Comment by michelengelen** [comment]
@spirosoik on my local machine it works like a charm.
When we set the env variable it gets added to the current App state, like seen in this screenhost:

Screenshot 2021-08-20 at 10 49 58

We then access it with a helper function like this: `const featureFlagEnabled = getFeatureFlagValue(state, 'GlobalHeader') === 'true';`

It accesses the state and gets the property `FeatureFlag{key}`, in our case `FeatureFlagGlobalHeader`.

The only case this is not working is when the feature flag is not set (or the flag is set, but not stored in the app state correctly).

I will try to have a look at the state on the server.

**Comment by spirosoik** [comment]
@michelengelen yes with env variable will definitely work the question is that we have this in Split and overall we are not setting feature flags static for the purpose to be dynamic through split. So something goes wrong with Split I think and it doesn't get the right value.

**Comment by spirosoik** [comment]
In practice all these values are coming from Split on purpose. I don't see global header though at all, like it's not there. Could you rebase just to be sure there is no issue with a wrong version of server? (wondering

Screenshot 2021-08-19 at 9 54 52 PM

**Comment by michelengelen** [comment]
WOW ... that rebase is messy for some reason. :/

**Comment by michelengelen** [comment]
@andrewbrown00 ready for review. Test server is running.

cc/ @spirosoik

**Comment by spirosoik** [comment]
so it was rebase issue

**Comment by michelengelen** [comment]
probably yes

**Comment by andrewbrown00** [comment]
@michelengelen I'm getting the white screen on this PR now. I made sure my cache was clear and tried in Incognito, but no luck. Are you seeing the white screen as well?

cc @spirosoik

**Comment by michelengelen** [comment]
well, not on this one ... strange though

Screenshot 2021-08-20 at 17 59 45

cc/ @spirosoik

**Comment by andrewbrown00** [comment]
@michelengelen this is what I'm seeing. I also include the `console` to see if it helps.

image

**Comment by michelengelen** [comment]
@andrewbrown00 and @jgilliam17 ... I just rebased this branch onto the current master, which incorporates the fix for the "white screen of death" bug. Gentle ping to re-review once the test server is up and running again.

**Review by matthewbirtch — APPROVED** [review]
I think this works @michelengelen. We will likely want to revisit whether we give the product brand an active/hover state, but we can look at that post 6.0

**Comment by matthewbirtch** [comment]
@michelengelen one thing that would be good: can you change the cursor to the pointer when hovering over the product brand?

**Comment by michelengelen** [comment]
> @michelengelen one thing that would be good: can you change the cursor to the pointer when hovering over the product brand?

@matthewbirtch done! 👍🏼

File: components/global/product_switcher.tsx [hunk 1]
```diff
@@ -142,6 +143,7 @@ const ProductSwitcher = (): JSX.Element => {
                         aria-label='Select to open product switch menu.'
                     />
                     <ProductSwitcherTip/>
+                    <ProductBranding onClick={handleClick}/>
```
Comment (crspeller) [inline components/global/product_switcher.tsx:146]: Maybe we can make the ProductSwitcherContainer handle the click so we don't have to pass this to two components?

**Review by crspeller — CHANGES_REQUESTED** [review]

File: components/global/product_switcher.tsx (hunk 1 above)
Comment (michelengelen) [reply components/global/product_switcher.tsx:146]: i did this to prevent closing of the menu when the divider is being clicked

Comment (michelengelen) [reply components/global/product_switcher.tsx:146]: oh, and it is easier to handle the action on the same level as the state for the opening/closing of the menu sits

**Comment by michelengelen** [comment]
@deanwhillier and @crspeller kind reminder to re-review

File: components/global/product_switcher.tsx (hunk 1 above)
Comment (michelengelen) [reply components/global/product_switcher.tsx:146]: sry ... my previous comments were for a different kind of component ... I mixed those up 🙇🏼‍♂️ 

But the same kinda goes for this as well: We don't want the space to be clickable, but only those that are visually indicating a potential action and by passing it to the container we would have the effect that the whole area is clickable

Comment (crspeller) [reply components/global/product_switcher.tsx:137]: But the only two things in the area are the two things we want clickable. I think it would be a bug if there was a tiny space between them that was unclickable. Presumably we want the whole area to eventually highlight not just the grid button?

**Comment by crspeller** [comment]
@michelengelen Not tally specific to this PR but you might want to read https://mattermost.com/blog/submitting-great-prs/ it's a great read and you might want to look closely at #4. Makes it much easier on reviewers 😄

**Comment by michelengelen** [comment]
> @michelengelen Not tally specific to this PR but you might want to read https://mattermost.com/blog/submitting-great-prs/ it's a great read and you might want to look closely at #4. Makes it much easier on reviewers 😄

thanks for the link ... I did not know of that. I rebased solely because we recently had issues with spinwicks not getting the latest server changes when we pulled master into the working branch. This might be resolved, but I wanted to make sure that all changes are up to date.

FYI: this is not how i normally do that! :D

File: components/global/product_switcher.tsx (hunk 1 above)
Comment (michelengelen) [reply components/global/product_switcher.tsx:137]: about the highlighting: that is something for UX to decide upon (although IMO you are right)
the clickable area: it depends ... right now there is only the 2 items with no space in between, so we can certainly apply it to the parent component

File: selectors/global_header.ts [hunk 2]
```diff
@@ -3,9 +3,12 @@
 
 import {GlobalState} from 'types/store';
 import {getFeatureFlagValue} from 'mattermost-redux/selectors/entities/general';
+import * as UserAgent from '../utils/user_agent';
+import * as Utils from '../utils/utils';
 
 export function getGlobalHeaderEnabled(state: GlobalState): boolean {
     const featureFlagEnabled = getFeatureFlagValue(state, 'GlobalHeader') === 'true';
+    const isMobileView = UserAgent.isMobile() || Utils.isMobile();
```
Comment (crspeller) [inline selectors/global_header.ts:11]: It looks like this unrelated code has appeared. Bad rebase?

**Review by crspeller — CHANGES_REQUESTED** [review]

File: selectors/global_header.ts (hunk 2 above)
Comment (michelengelen) [reply selectors/global_header.ts:11]: yep ... sadly! will remove

**Review by crspeller — APPROVED** [review]

File: components/global/product_branding/product_branding.tsx [hunk 3]
```diff
@@ -11,6 +11,7 @@ import {useProducts, useCurrentProductId} from '../hooks';
 const ProductBrandingContainer = styled.div`
     display: flex;
     align-items: center;
+    cursor: pointer;
```
Comment (deanwhillier) [inline components/global/product_branding/product_branding.tsx:14]: I'm wondering if this should maybe be relocated to `ProductSwitcherContainer` since the `onClick` handler is there so we can keep the functionality and the visual affordance for the functionality "together".

**Review by deanwhillier — CHANGES_REQUESTED** [review]
Apologies for the delayed review @michelengelen, I somehow lost track of this one. LGTM with one possible suggested tweak.

**Comment by michelengelen** [comment]
@deanwhillier feel free to merge once you re-review and approve! 🙇🏼‍♂️

**Review by deanwhillier — APPROVED** [review]
